tests/dummy/app/routes/application.js

```diff
@@ -10,8 +10,8 @@ export default Ember.Route.extend({
   },
 
   actions: {
-    update(newOrder) {
-      this.set('currentModel.items', a(newOrder));
+    update(model, newOrder) {
+      a(model).set('items', a(newOrder));
```

Comment:
I'm getting an error on that line when I do not wrap the model in an Ember.Array. It's probably not the best way to fix it, but the error wasn't very descriptive:
```
update@http://localhost:4200/assets/dummy.js:150:9
triggerEvent@http://localhost:4200/assets/vendor.js:35712:13
trigger@http://localhost:4200/assets/vendor.js:59036:9
Router.prototype.trigger@http://localhost:4200/assets/vendor.js:57760:9
EmberRouter<.send@http://localhost:4200/assets/vendor.js:35210:8
ActionHandler<.send@http://localhost:4200/assets/vendor.js:38901:9
TargetActionSupport<.triggerAction@http://localhost:4200/assets/vendor.js:42089:17
Component<.sendAction@http://localhost:4200/assets/vendor.js:50729:1
commit@http://localhost:4200/assets/vendor.js:75866:9
_tellGroup@http://localhost:4200/assets/vendor.js:76136:9
_complete@http://localhost:4200/assets/vendor.js:76221:7
Backburner.prototype.join@http://localhost:4200/assets/vendor.js:10799:18
run.join@http://localhost:4200/assets/vendor.js:27952:12
run.bind/<@http://localhost:4200/assets/vendor.js:28014:14
tryCatch@http://localhost:4200/assets/vendor.js:59478:16
invokeCallback@http://localhost:4200/assets/vendor.js:59490:17
publish@http://localhost:4200/assets/vendor.js:59461:11
@http://localhost:4200/assets/vendor.js:38429:7
Queue.prototype.invoke@http://localhost:4200/assets/vendor.js:11432:11
Queue.prototype.flush@http://localhost:4200/assets/vendor.js:11497:13
DeferredActionQueues.prototype.flush@http://localhost:4200/assets/vendor.js:11302:13
Backburner.prototype.end@http://localhost:4200/assets/vendor.js:10727:11
Backburner.prototype.run@http://localhost:4200/assets/vendor.js:10782:15
executeTimers@http://localhost:4200/assets/vendor.js:11164:7
updateLaterTimer/backburner._laterTimer<@http://localhost:4200/assets/vendor.js:11153:11
```